PR: https://github.com/odoo/odoo/pull/146268

From: 6d2ca901cfd75530178096efae90b91363d05931
From: rupa-odoo
Date: 2024-02-16 16:11:46

Structural Changes: 2.8
Total Changes: 9

[IMP] product, _*: improve products tooltip

_*= stock, website_sale, point_of_sale

Before these commit:
-> There is a tooltip 'favorite' displayed for the star icon which is not more
explicit about the star icon.

After these commit:
-> Changed the tooltip displayed for the star icon to:
"add to favorites / remove from favorites"

task-3476798

closes odoo/odoo#146268

Related: odoo/enterprise#52762
Related: odoo/upgrade#5441
Signed-off-by: Xavier Bol (xbo) <xbo@odoo.com>

================================= pseudo patch: =================================

--- a/addons/product/models/product_product.py
+++ b/addons/product/models/product_product.py
@@ -17,7 +17,7 @@ class ProductProduct(models.Model):
     _description = "Product Variant"
     _inherits = {'product.template': 'product_tmpl_id'}
     _inherit = ['mail.thread', 'mail.activity.mixin']
-    _order = 'priority desc, default_code, name, id'
+    _order = 'is_favorite desc, default_code, name, id'
 
     # price_extra: catalog extra value only, sum of variant extra attributes
     price_extra = fields.Float(

--- a/addons/product/models/product_template.py
+++ b/addons/product/models/product_template.py
@@ -17,7 +17,7 @@ class ProductTemplate(models.Model):
     _name = "product.template"
     _inherit = ['mail.thread', 'mail.activity.mixin', 'image.mixin']
     _description = "Product"
-    _order = "priority desc, name"
+    _order = "is_favorite desc, name"
 
     @tools.ormcache()
     def _get_default_category_id(self):
@@ -146,10 +146,7 @@ class ProductTemplate(models.Model):
 
     product_tooltip = fields.Char(compute='_compute_product_tooltip')
 
-    priority = fields.Selection([
-        ('0', 'Normal'),
-        ('1', 'Favorite'),
-    ], default='0', string="Favorite")
+    is_favorite = fields.Boolean(string="Favorite")
 
     product_tag_ids = fields.Many2many(
         string="Product Template Tags",
